Cami2-assembly-tutorial (#3293)
galaxyproject/training-material: topics/metagenomics/tutorials/cami2-assembly/tutorial.md

---
layout: tutorial_hands_on

title: "Reproducing Critical Assessment of Metagenome Interpretation assembly challenge on marine dataset with Galaxy"
zenodo_link: ""
questions:
  - "How to reproduce CAMI challenge?"
  - "How to add/update tool in Galaxy?"
  - "How to benchmark results of CAMI challenge?"
objectives:
  - "Be familiar with CAMI challenge"
  - "Be able to select one challenge and benchmarking datasets"
  - "Be able to upload benchmarking datasets into Galaxy"
  - "Be familiar with Galaxy computational resources"
  - "Be familiar with existing tools in Galaxy"
  - "Be familiar with databases / reference genomes that are available in Galaxy"
  - "Be able to add/update the tool in Galaxy"
  - "Be able to choose assembling tool based on dataset metadata"
  - "Perform assembly with Flye, Megahit, Abyss, MetaSPAdes tools"
  - "Produce a benchmarking analysis of these assemblies with Quast, Bowtie2, Map with Minimap2, Samtools, and MultiQC tools"
  - "Create plots with Python to compare results"

> time_estimation: "40H00M"

Author: Thank you for the comment. Do you have any advice how can I count this number?

> time_estimation: "40H00M"

Reviewer: It's usually "how long does it take to teach/follow this tutorial from start to finish", if some of the assembly steps take that long, then, that can be accurate!

> time_estimation: "40H00M"

Author: > It's usually "how long does it take to teach/follow this tutorial from start to finish", if some of the assembly steps take that long, then, that can be accurate! I was just surprised at the big number :)

Then I would put more since the slowest assembler took 41h and after that Quast should be run and it took 3h. And it was just the pure time of one assembler (out of 9) and Quast (there were other tools run though).

    * Marine
        * Simulated short read and long read shotgun metagenome data from samples at different seafloor locations of a marine environment
    * Plant-associated
        * Simulated short read and long read shotgun metagenome data from samples taken from a plant rhizosphere environment
    * strain madness
        * Simulated long read and short read shotgun metagenome data, including a large amount of strain-level variation
* a pathogen detection challenge dataset was offered, based on a clinical metagenome sample of blood from a critically ill patient with hemorrhagic fever of unknown cause.

During the challenges, Datasets for the challenge were available for download only to participants via the [CAMI portal](https://data.cami-challenge.org/). After the challenges, all CAMI benchmark datasets were made available with digital object identifiers (DOIs) ([Table 2](https://www.nature.com/articles/s41596-020-00480-3/tables/2)) and the genomic data are now in public sequence repositories such as the National Center for Biotechnology Information (NCBI) to be used for further benchmarking in the field.

> Different tools were run with different parameters on the various datasets. Participants ran their preferred tools with their preferred set of parameters and  submitted their results along with either a Docker container containing the complete workflow, a bioconda script or a software repository with detailed installation instructions, specifying all parameter settings and reference databases used. A total of 5,002 submissions were received from 30 external teams and CAMI developers for the four challenge datasets. Following that, the CAMI developers evaluated the results using standardised metrics and then made sense of the different results described in [Meyer et al, 2020](https://www.biorxiv.org/content/10.1101/2021.07.12.451567v1)

Reviewer: ```suggestion
Different tools were run with different parameters on the various datasets. Participants ran their preferred tools with their preferred set of parameters and  submitted their results along with either a Docker container containing the complete workflow, a bioconda script or a software repository with detailed installation instructions, specifying all parameter settings and reference databases used. A total of 5,002 submissions were received from 30 external teams and CAMI developers for the four challenge datasets. Following that, the CAMI developers evaluated the results using standardised metrics and then made sense of the different results described in [Meyer et al, 2020](https://www.biorxiv.org/content/10.1101/2021.07.12.451567v1)
```

Here the meyer et al would be best replaced with a citation. 

1. Take the doi, and turn it into a bibliography entry via doi2bib.org
2. and then add that to your bib file 
3. replace the meyer et al 2020 with `{% cite ... %}`.

Author: ok

Three “challenge” datasets were offered: 

* marine, 
* strain madness, and 
* plant associated data. 

Two types of assemblies were assessed: single-sample assembly, i.e. assembly of reads from only one sample at a time, and co-assembly, i.e. assembly of all reads from several samples together. For the sake of comparison, gold standard assemblies, i.e. the correct result which can be used to compare with and benchmark other methods’ results, were created with all regions covered by at least one read and were provided only after the challenges.

Participants provided 155 submissions and 20 assembler versions of 10 assembling tools to be evaluated during this challenge. The different tools performed differently while measuring various metrics. By CAMI2 developers there are provided [plots](https://www.biorxiv.org/content/biorxiv/early/2021/07/12/2021.07.12.451567/F1.large.jpg?width=800&height=600&carousel=1) of benchmarking analyses of different tools depending on different statistics.

![Assembly tools plot from CAMI2](../../images/cami2/assembly-tools-plots.jpg){:width="70%"}

> Overall, GATB[[3]](https://pubmed.ncbi.nlm.nih.gov/24990603/) ranked best on the strain madness data across most metrics (mismatches, misassemblies, strain recall, and strain precision), while HipMer[[4]](https://ieeexplore.ieee.org/abstract/document/8665813) [[5]](https://ieeexplore.ieee.org/abstract/document/7832788) [[6]](https://www.nature.com/articles/s41598-020-67416-5) on the plant-associated data. Compared to the first CAMI challenge A-STAR[ [7]](https://www.a-star.edu.sg/) considerably improved in genome fraction for the strain madness data. The other leader SPAdes [[8]](https://pubmed.ncbi.nlm.nih.gov/22506599/) [[9]](https://genome.cshlp.org/content/27/5/824?ijkey=fef5a6b24fb0024b39509480f85afaa81724d3e8&keytype2=tf_ipsecsha) was not introduced in the CAMI1 challenge [[1]](https://www.nature.com/articles/nmeth.4458) and performed well in CAMI2. For the type of assembly, the different tools performed differently. Single-sample assembly was done well by all assemblers. HipMer [[4]](https://ieeexplore.ieee.org/abstract/document/8665813) [[5]](https://ieeexplore.ieee.org/abstract/document/7832788) [[6]](https://www.nature.com/articles/s41598-020-67416-5) performed better on single samples as well as for pooled samples.

Reviewer: Here all of these references need to be fixed. For references please use the proper referencing format (bib entries in tutorial.bib, `{% cite referenceName %}` in the text.

`A-STAR[ [7]](https://www.a-star.edu.sg/)` is a bit of an odd reference, is there a paper that could be used here? If it's just a link and not intended to be a reference, you should do something like `the first CAMI challenge [A-STAR](https://...)`, generally it's best to make the link part of the text itself, not separate from the text.

#### Genome binning challenge

Genome binning  clusters or classifies sequences or contigs into bins representing genomes. 

It takes as input the output of assembly. Two types of short read assemblies were given as inputs for CAMI2 [[2]](https://www.biorxiv.org/content/10.1101/2021.07.12.451567v1): MEGAHIT [[10]](https://www.biorxiv.org/lookup/external-ref?access_num=25609793&link_type=MED&atom=%2Fbiorxiv%2Fearly%2F2021%2F07%2F12%2F2021.07.12.451567.atom) assembly and Gold Standard Assembly. Both types for strain madness, marine, and plant-associated datasets, so 6 possible datasets in total.

Participants submitted 95 results and 10 binning tools (overall, 18 binner versions). The results were evaluated based on .

 

* the average purity of bins, 
    * The average purity is the fraction of correctly assigned base pairs for all assignments to a given bin averaged over all predicted genome bins, where unmapped genomes are not considered. 
    * This value reflects how trustworthy the bin assignments are on average. 
* completeness of genomes recovered,
* Adjusted Rand Index which quantifies binning performance for the overall dataset.

>     * According to the [paper](https://www.biorxiv.org/lookup/google-scholar?link_type=googlescholar&gs_type=article&author[0]=F.+Meyer&title=AMBER:+Assessment+of+Metagenome+BinnERs&publication_year=2018&journal=Gigascience&volume=7) [[11]](https://academic.oup.com/gigascience/article/7/6/giy069/5034950?login=true), genome binners generate groups or clusters of reads and contigs for a given dataset. Instead of calculating performance metrics established with a bin-to-genome mapping, the quality of a clustering can be evaluated by measuring the similarity between the obtained and correct cluster partitions of the dataset, corresponding here to the predicted genome bins and the gold standard contig or read genome assignments, respectively. Two contigs or reads of the same genome that are placed in the same predicted genome bin are considered true positives TP. Two contigs or reads of different genomes that are placed in different bins are considered true negatives TN. The Rand index ranges from 0 to 1 and is the number of true pairs, TP + TN, divided by the total number of pairs. However, for a random clustering of the dataset, the Rand index would be larger than 0. The adjusted Rand index (ARI) corrects for this by subtracting the expected value for the Rand index and normalising the resulting value, such that the values still range from 0 to 1.

Reviewer: I won't comment on every single reference issue, but, the comments above apply to all of the references throughout this tutorial

>
{: .details}

> ### {% icon details %} CAMI2 Genome binning tools comparison
>
> <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=836882193&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>
>
{: .details}

> ### {% icon details %} CAMI2 Taconomic binning tools comparison
>
> <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=794114188&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>
>
{: .details}

> ### {% icon details %} CAMI2 Taxonomic profiling tools comparison
>
> <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=2127016351&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>
>

> {: .details}

Reviewer: Linking all of the spreadsheets via iframes feels a bit odd, I think it might be a bit nicer to collect all of those in a details box as links, something like this instead of the iframes. Would that be ok for you? (note, links aren't correct here.)

```
> ### {% icon details %} CAMI2 Details
>
> - [CAMI2 Assembly tools comparison](https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml)
> - [CAMI2 Genome binning tools comparison](https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml)
> - [CAMI2 Taconomic binning tools comparison](https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml)
{: .details}
```

### Datasets description

##### Toy datasets

As in the first CAMI challenge metagenome “toy” (or “practice”) benchmark datasets were created from public genomes and provided together with the standard of truth before the challenges, to enable contest participants to familiarise themselves with data types and formats. These included

*  a 49-sample simulated metagenome data from Human Microbiome data to represent five different body sites of the human host, namely gastrointestinal tract, oral cavity, airways, skin and urogenital tract.

> ### {% icon details %} Human microbiom dataset details
>
> <iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=503333420&amp;single=true&amp;widget=true&amp;headers=false" width="900" height="600" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>
>

> {: .details}

Reviewer: ```suggestion
*  a [49-sample simulated metagenome data from Human Microbiome data](https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=503333420&amp) to represent five different body sites of the human host, namely gastrointestinal tract, oral cavity, airways, skin and urogenital tract.

Reviewer: Here condensing the link from a separate box into just a hyperlink on the dataset will make it a bit more concise and readable I think.

> *  a 64-sample simulated metagenome data from the guts of different mice, vendors and positions in the gut. (**[https://openstack.cebitec.uni-bielefeld.de:8080/swift/v1/CAMISIM_MOUSEGUT](https://openstack.cebitec.uni-bielefeld.de:8080/swift/v1/CAMISIM_MOUSEGUT)**)

Reviewer: ```suggestion
*  a [64-sample simulated metagenome data from the guts of different mice, vendors and positions in the gut.](https://openstack.cebitec.uni-bielefeld.de:8080/swift/v1/CAMISIM_MOUSEGUT)
```

Here the url itself isn't so important, it's not important that users are able to see the URL, we should just again link the thing you're talking about to the correct location.

Both include  5 Gb long paired-end reads (Pacific Biosciences, variable length with a mean of 3000 bp) and 5 Gb short paired-end reads (Illumina HiSeq2000, 150 bp).

##### Challenge datasets

For the assembly challenge itself, 4 “challenge” datasets were offered: 

* clinical pathogen detection,
* marine, 
* strain madness, 
* plant associated data. 

###### Clinical Pathogen detection

The clinical pathogen detection dataset, a short-read metagenomic sequencing dataset of a blood sample from a patient with hemorrhagic fever of unknown cause, was excluded because it is provided only for the last challenge: identify a causal pathogen together with further pathogens.

> > ### {% icon details %} Clinical Pathogen detection dataset links

Reviewer: same here, I think, just link text with the url you're interested in, rather than using details boxes with iframes. (This same comment applies to the rest of the datasets you discuss / details boxes with iframes.)

#### Download data from CAMI2 resources.

Since data is of large size we cannot just download them to our computer by clicking the button download. Additionally, data cannot be downloaded at once, datasets are divided by samples - one file per one sample. There are also different files for long and short reads of the same sample.

It is possible to download datasets using DOI links from [Table 2](https://www.nature.com/articles/s41596-020-00480-3/tables/2) CAMI Tutorial [[20]](https://www.nature.com/articles/s41596-020-00480-3)

| CAMI benchmark dataset                                               | DOI                                              |
|--------------------------------------------------------------------- |------------------------------------------------- |
| CAMI I: low, medium, high complexity, and ‘toy’ datasets             | 10.5524/100344                                   |
| CAMI II: human microbiome project and mouse gut toy datasets         | 10.4126/FRL01-006425518; 10.4126/FRL01-006421672 |

> | CAMI II: marine, strain madness, rhizosphere, and pathogen detection | 10.4126/FRL01-006425521                          |

Reviewer: Here it'd be really nice to link those DOIs! Either using `{% cite %}` (+ add to bib file) or hyperlinking directly.

> ### {% icon hands_on %} Hands-on: Get dataset with DOI
> 
> 1. Use the table above to find unique identifier (DOI) of dataset you need

> > 2. Use [https://search.datacite.org/](https://search.datacite.org/) to search for data via DOI, and click the link that was found for the DOI. 

Reviewer: ```suggestion
> 2. Use [DataCite](https://search.datacite.org/) to search for data via DOI, and click the link that was found for the DOI. 
```

generally we avoid using URLs in the `[]` portion of the link, they're a bit unattractive to read.

> 3. On the page of dataset there is the link provided to download dataset -  for marine dataset all samples available via [https://frl.publisso.de/data/frl:6425521/marine/](https://frl.publisso.de/data/frl:6425521/marine/)

> > 4. Download datasets on your computer

Reviewer: If the datasets are available via URL, this should not be a step, they should upload directly to Galaxy.

A great option here is upload [via the rule builder](https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/upload-rules/tutorial.html)

There we often have students create tables (or provide tables for them) and use those to upload + properly tag/name datasets.

> > 
> > 1. Install Java 8 on your computer.
> > 2. Create an account on[ CAMI portal](https://data.cami-challenge.org/)
> > 3. Download the [camiClient.jar](https://data.cami-challenge.org/camiClient.jar)
> > 4. Get text-file with links for dataset from [CAMI portal](https://data.cami-challenge.org/). You can download this file in the section “2nd CAMI Challenge Marine Dataset”
> > 5. You can download data with CAMI client using the following command line in the terminal:
> > 
> >     ```
> > java -jar camiClient.jar -d <linkfileLocation> <targetDirectory> -p PATTERN
> > ```
> {: .hands_on}
> 
{: .details}

#### Upload data into Galaxy

There are different ways to upload data into the Galaxy. In case of a not very large dataset it’s more convenient to upload data directly from your computer to Galaxy. In this tutorial we use a large dataset, thus, we use FTP server for this purpose.

> ### {% icon hands_on %} Hands-on: Upload data into Galaxy with FTP
>
> 1. Create new history
>
>    {% snippet faqs/galaxy/histories_create_new.md %}
>
> 2. Upload datasets into FTP Galaxy server
>
>    {% snippet faqs/galaxy/datasets_upload_ftp.md %}
>

> > 3. Press “upload data” -> “choose remote files” -> “FTP directory” and upload your data to your history from FTP server

Reviewer: The rule builder would remove these steps which would make it *much* easier for folks to follow this tutorial, FTP is a big complication for users.

Author: Do you mean, it would be better if I remove these steps from the tutorial?

Reviewer: Yes, remove, and replace with the Rule Builder to upload by URLs. E.g. https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html#import-data-from-urls they provide a text table that users can copy-paste into Galaxy, and below is a set of instructions for using the rule based uploader to turn those into datasets in a history.

Author: Thank you, I just learned this way, and it's way more convenient than ftp. However, in case of CAMI2 datasets links, every time you should download the file with temporary links for datasets. They are different every time and have an expire date. So, maybe it's better to ask a Galaxy user to create this text table with links by their own and then to use it in Rule Builder?

Reviewer: Oh wow that's difficult! It makes sense for the challenge though.

1. Yes, I think that's a good idea, show them an example of the table and then tell them to construct their own or so.
1. are these datasets restricted, is there something like a "you may not share these publicly" applied? If not we could put a static set of them on zenodo so that people could follow the tutorial reproducibly?

Author: I changed the tutorial according to the 1st option.

{: .hands_on}

#### Preprocessing data

We got the marine data in our history, with one file per sample (10 samples for long reads and 10 samples for short reads). We need to prepare our data.

> 1. We should organise our data into collections. For deeper learning about dataset collections follow {% cite noauthor_galaxy_nodate-1 %} tutorial.

Reviewer: Here you're referencing another tutorial, citing is not proper in this case, instead just link to it (And remove the associated citation.)

```suggestion
1. We should organise our data into collections. For deeper learning about dataset collections [follow our tutorial]({% link topics/galaxy-interface/tutorials/collections/tutorial.md %}).
```

Author: ok, thanks

2. These files consist of interleaved paired-end reads, i.e. forward and reverse reads are all together in one single file and not 2 as often. This can be a problem for some assemblers as not every assembling tool can take interleaved paired-end reads as input.For co-assembly of all samples, some assemblers expect one file with all samples  combined together.

> ### {% icon hands_on %} Hands-on: Preprocessing data
>
> 1. Take files for short reads for all samples
> 2. Create a collection with all samples- We now have a collection with all files. Let's now combine all sample files into one file

> > 3. Combine all samples from collection into one file -  follow the section “[Collapse data into a single dataset](https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html#collapse-data-into-a-single-dataset)” in {% cite noauthor_galaxy_nodate-1 %}

Reviewer: ```suggestion
> 3. Combine all samples from collection into one file -  follow the section *Collapse data into a single dataset* in the [collections tutorial]({% link topics/galaxy-interface/tutorials/collections/tutorial.md%}#collapse-data-into-a-single-dataset)
```

Author: ok

> 
>      We can now split forward and reverse reads into 2 files
> 

> > 4. **FASTQ splitter** {% icon tool %} with following parameters:

Reviewer: If you could use the new tool linking syntax it would be really nice for users! Then they could use "Tutorial Mode"

We have a [guide on how to do it](https://training.galaxyproject.org/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html#tool-links), but in short just switch to `{% tool [Your Tool](toolshed.g2.bx.psu.edu/repos/..../your-tool/1.0.0+galaxy0) %}`

Author: ok, good idea

To update or fix a tool, we need to change its XML. Tools in a Galaxy server are installed by the administrators of this Galaxy server using the Galaxy ToolShed. The Galaxy ToolShed is like an AppStore with the XML for all possible Galaxy tools. So to update a tool, we need to change its XML in the ToolShed. 
But usually, XML of tools are developed collaboratively inside GitHub repositories and then pushed to the ToolShed. So we need to find the development repository of the Galaxy tool to update it. 

> ### {% icon hands_on %} Hands-on: Finding the development repository of a tool to update
> 
> 1. Open the fastqc splitter in Galaxy
> 2. Click on “See in the ToolShed” on the top right of the tool form. You will be redirected to the tool on the ToolShed
> 3. Follow the link in “**Development repository”**
> 4. **You have now the tool wrapper, i.e. its XML**
> 5. Once we have the development repository setup, we can update the tool. We recommend using Planemo, the Galaxy software development kit, to help the process. More information is in {% cite noauthor_building_nodate %}.
{: .hands_on}

> > ### {% icon hands_on %} Hands-on: Update the tools

Reviewer: I don't think this is a hands-on, right? The tool is already updated, you're just documenting what was done, right? If so, please put it in a `details` box or so, rather than a hands-on

Author: I agreed

>    2. Select `Add / Modify Column Definitions`
>    3. Define the column with Identifier
>       1. Click `Add Definition` button
>       2. Select `List Identifier(s)`
>       3. Select `A` column
>    4. Define the column with URL
>       1. Click `Add Definition` button
>       2. Select `URL`
>       3. Select `B` column
>    5. Click on `Apply`
> 10. Add the name `Short reads` to the collection
> 11. Click on `Upload`
>
>     > <comment-title></comment-title>
>     > The process will take time.
>     {: .comment}
>
> 12. Inspect the generated collection
>
{: .hands_on}

> <question-title></question-title>
>
> 1. What is the format of the file for each sample?
> 2.
>
> > <solution-title></solution-title>
> >
> > 1.
> >
> {: .solution}
>
{: .question}

We got the marine data for short-reads in our history. Let's do the same for long-reads.

> <hands-on-title>Upload long-read data into Galaxy with Rule Builder</hands-on-title>
>
> 1. Click the upload icon toward the top left corner.
> 2. Click on `Rule-based` on the top right
> 3. Select `Collection(s)` in **Upload data as**
> 4. Paste the following tabular data right into the textbox:
>
>     ```text
>     SampleID	URL

> >     Long read sample 0	https://frl.publisso.de/data/frl:6425521/marine/long_read/marmgCAMI2_sample_0_reads.tar.gz

Reviewer 2: These links do not work: it is tar.gz with subfolders in. We will put the data in the shared data library

>
>    > <comment-title></comment-title>
>    > While working on this project we realised that both outputs (forward and reverse) had the same name `FASTQ splitter on data X`, where X is the internal number of data in this Galaxy history, with the version of FASTQ splitter we used
>    > Since we have run FASTQ splitter on one dataset and got 2 outputs, as a result filenames of both outputs were the same.
>    >
>    > Why is it not good?
>    >
>    > It can be mistaken by assembler (or any other tool where we want to use FASTQ splitter output files) because it will not know which file is reverse and which is forward.
>    >
>    > We updated the Galaxy tool to fix this issue
>    {: .comment}
>
> 6. Repeat the steps for long reads
{: .hands_on}

We now have collections for both short and long reads.

# Select and prepare the tools to run

Based on {% cite meyer2022critical %}, {% cite Meyer2021 %} and {% cite Meyer2021_tutorial %}, we can compare tools on a set of metrics to select the one to use for an analysis but also here to run the challenge:

> Tool | Genome fraction (%) | Mismatches per 100 kbp | Misassemblies | NGA50 | Strain recall | Strain precision

Reviewer 2: It would be good to add before a short explanation of the different columns

Gold Standard Assembly (GSA) | 76.9 | 0 | 0 | 682,777 | 54.9% (upper bound) | 100
ABySS ({% cite jackman2017abyss %}) | Very accurate mean fraction (<1% divergence) |  | The fewest misassemblies | 100% precision. The highest strain precision (100%) for the unique genome
Ray ({% cite Boisvert2012 %}) |  |  |  |  |  | 100% precision. The highest strain precision (100%) for the unique genome

> A-STAR | A-STAR excelled in terms of genome fraction on marine and strain madness data sets. A-STAR improved the genome fraction to 44.1% on the marine dataset. On marine common genomes, A-STAR (26.7%) achieved the highest genome fractions. For unique genome, A-STAR provided the most complete assemblies (55.3% genome fraction). A-STAR partially recovered 102 (78%) of 131 16S gold standard sequences. | More mismatches than others: 773/100 kb | More misassemblies than others | | 2nd highest: 7.5% recall  | 2nd highest: 69.4% precision

Reviewer 2: Could you simply the content of the cells in this table and the ones in the detail box?

> OPERA-MS [25] | There were selected 50 unique, public genomes present as a single contig in the gold standard and with annotated 16S sequences. The hybrid assembler OPERA-MS recovered one of the most complete 16S sequences (mean recovered gene fraction 47.1%). For the unique genome, OPERA-MS has an exceptional average NGA50 (187,083, 75% of the gold standard NGA50). | | The most contiguous assemblies were provided by the hybrid assembler OPERA-MS for the marine data, with an average NGA50 of 28,244 across genomes. |

Reviewer 2: Please add the correct citations to the tool

> SPAdes |
> Flye |
>
>
> **Plant-associated**
>
> Genome fraction (%) | Mismatches per 100 kbp | Misassemblies | NGA50 | Strain recall | Strain precision
> GSA |
> ABySS |
> Ray |
> A-STAR |
> OPERA-MS |  |  | High number of mismatches |  | Extremely small strain recall | OPERA-MS has the lowest precision
> MEGAHIT |
> HipMer (performed best across metrics) | HipMer, HipMer2, and Flye have the best coverage. | HipMer has less mismatches per 100 kbp compared to other assemblers. Followed by HipMer2 |  |  | HipMer and HipMer2 have the highest strain recall |
> GATB |
> SPAdes |  |  | SPAdes has the lowest number of misassemblies |
> Flye (the second place) | HipMer, HipMer2, and Flye have the best coverage. |  | Flye_hybrid has higher number of misassemblies than others |  |  | Flye, HipMer, amd HipMer2 have the highest precision
{: .details}

Using the described metrics, the different tools were evaluated in the CAMI paper and aggregated in tables (Supplementary Tables 3-7) from {%cite tutorialMeyer2021%}.

> In these tables there are also ranking scores of the tools shown for every statistic as well as overall ranking scores. Overall, ranking scores for every dataset are computed as a sum of all ranking scores across metrics. The average ranking score of both datasets are calculated as weighted average sum of ranking for both datasets. We created [a table showing all ranking results from previous tables](https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml?gid=455354696)

Reviewer 2: Could you add the table directly in the tutorial? Thanks

We summed up ranking scores for all methods, common genomes, and unique genomes to extract the following  priority list for tools

**Marine** dataset

_With tool versions_
1. HipMer
2. metaSPAdes_v3.13.1
3. metaSPAdes_v3.13.0
4. ABySS
5. Ray-Meta
6. Megahit_v1.1.2
7. SPAdes_v3.14-dev

_Without tool versions_
1. HipMer
2. **metaSPAdes**
3. **ABySS**
4. **Ray-Meta**
5. **Megahit**

**Strain madness** dataset
_With tool versions_
1. HipMer
2. Megahit_v1.1.2
3. SPAdes_v3.14-dev
4. OPERA-MS
5. Megahit_V1.2.7

_Without tool versions_
1. HipMer
2. **Megahit**
3. **SPAdes**
4. **OPERA**

**Plant-associated** dataset

There are no certain ranking tables among Supplementary tables for plant-associated dataset. However,  in {%cite Meyer2021%} there is information related to tools performance on plant-associated dataset. We created the priority list of tools for the plant-associated dataset.

1. (Meta)HipMer
2. **(meta)Flye**

> 3. **(meta)SPAdes**

Reviewer 2: It would be nice to show that maybe as a table

> Since in this tutorial we have decided to focus on marine dataset it would be reasonable to reproduce CAMI2 assembly challenge using HipMer, metaSPAdes, ABySS, Ray-Meta, Megahit assemblers which performed better. As we know from our [comparison Galaxy and CAMI2 analysis](https://docs.google.com/spreadsheets/d/e/2PACX-1vQgJr3J-IyVy9IkXS9W-RZcV83Tr6f7RusG_97QwgpW2dFdCXUMroROIhy8gKjPcUgISFXW9NQwOzzK/pubhtml), metaSPAdes, ABySS, Megahit tools are available in Galaxy while Ray-Meta and HipMer are not.

Reviewer 2: Reference to a table above
